docs/docs/how-to/plugins-and-themes/adding-gatsby-image-support.md

```diff
@@ -0,0 +1,261 @@
+
+
+  return generateImageData(imageDataArgs)
+}
+
+```
+
+### Add the resolver
+
+You should register the resolver using the [`createResolvers` API hook](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-node/#createResolvers). `gatsby-plugin-image/graphql-utils` includes an optional utility function to help with this. It registers the resolver with all of the base arguments needed to create the image data, such as width, aspect ratio, layout and background color. These are defined with comprehensive descriptions that are visible when your users are building queries in GraphiQL. You can pass additional arguments supported by your plugin, for example image options such as quality. However if you want complete control over the resolver args, then you can create it yourself from scratch. We recommend keeping the args similar to the default, as this is what users will be expecting, and it means you benefit from the plugin documentation. At a minimum, you should always expose `layout`, `width` and `height` as args.
+
+The arguments:
+
+- `resolverFunction`: the resolver function that you created at step 1. It receives the node and the arguments and should return the image data object.
+- `additionalArgs`: an object defining additional args, in the same format used by [Gatsby Type Builders](https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#gatsby-type-builders)
+
+For example, to add a `gatsbyImageData` resolver onto a `ProductImage` node that you have previously defined:
+
+```js:title=gatsby-source-example/gatsby-node.js
+// Note the different import
+import { getGatsbyImageResolver } from "gatsby-plugin-image/graphql-utils"
+
+export function createResolvers({ createResolvers }) {
+  createResolvers({
+    ProductImage: {
+      // loadImageData is your custom resolver, defined at step 2
+      gatsbyImageData: getGatsbyImageResolver(loadImageData, {
+        quality: "Int",
+      }),
+    },
+  })
+}
+```
+
+## Adding a custom image component
+
+If you have a URL-based image API, you can create a custom image component that wraps `<GatsbyImage />` and displays images generated at runtime. If you have a source plugin, this could accept your native image object, but it could equally just take a base URL and generate the image based on that. This is a good solution for image CDNs that aren't handling their own CMS data, and can generate a transformed image from just a source URL and dimensions.
+
+There are three steps to create a custom image component:
+
+1. [Create your URL builder function](#create-your-url-builder-function)
+2. [Create your image data function](#create-your-image-data-function)
+3. [Create your wrapper component](#create-your-wrapper-component) (optional)
+
+### Create your URL builder function
+
+This is similar to the `generateImageSource` described above, but just returns a URL string. This is an example for the same image host:
+
+```js
```

Comment:
Should this go in a particular file?

Reply:
Not really. It could be in gatsby-node, but is normally in another file